webapp-django/crashstats/crashstats/tests/test_utils.py

```diff
@@ -523,8 +524,8 @@ def test_unicode_writer(self):
         ])
 
         result = out.getvalue()
-        assert isinstance(result, str)
-        u_result = unicode(result, 'utf-8')
+        assert isinstance(result, string_types)
```

Comment:
This looked wrong to me because the original code is testing for `str`--not `basestring` which is `str` + `unicode`. I think this should be something like:

```
assert isinstance(result, six.binary_type)
```

After writing that, I got to wondering what the rest of the test is doing such that it's asserting something is a bytes here because that's odd. I read through the test and looked up some things like what does `StringIO.getvalue()` return in Python 2 and Python 3. I think we need to make other changes to this test.

The problem is that `StringIO` is being used to hold a stream of bytes--not characters. In Python 2, that's a `str` and `StringIO`. In Python 3, that's `bytes` and `BytesIO`. So we need to use one set of things in Python 2 and the other set in Python 3.

So I think this probably needs to be something like this:

```
def test_unicode_writer(self):
    if six.PY2:
        out = StringIO()
    else:
        out = BytesIO()
    writer = utils.UnicodeWriter(out)
    writer.writerow([
        'abc',
        u'\xe4\xc3',
        123,
        1.23,
    ])
    result = out.getvalue()
    assert isinstance(result, six.binary_type)
    u_result = result.decode('utf-8')
    # and so on
```

I'm pretty sure this is true, but it's hard to know for sure because we don't have this test running in the Python 3 harness, yet. Thus we can't make some changes and see whether they work in Python 2 and Python 3. In order to do that, we need to make a bunch of other changes first which we can't do until we've fixed linting issues. Also (and I bring this up purely as a "haha--wouldn't it be funny if ..." exercise) it's entirely possible `UnicodeWriter` doesn't work at all in Python 3. So much fun!

Given that we can't run this in the Python 3 test harness and know whether it's correct or not, I vote we slink back to a happier place and make the smallest change to move forward that we know works in Python 2 because that's what's running in production. I think the simplest fix is a minor tweak on what you have:

```
assert isinstance(result, six.bytes_type)
```

Then we leave the rest of this "hot mess" to future us. Because we like us and don't want future us disparaging current us, we should leave ourselves a note along the lines of:

```
# NOTE: This probably needs to be a StringIO in Python 2 and a BytesIO in Python 3
# assuming UnicodeWriter even works in Python 3.
```

I think that's right.